### PR TITLE
Update Building-Projects.rst

### PR DESCRIPTION
**Pull request by sbc100**
Remove most references to `.bc` or bitcode in general in favor
of just `object file` and `.o`.

Remove old advise about library files that no longer holds.

### CODE REVIEW

File: site/source/docs/compiling/Building-Projects.rst [hunk 1]
```diff
@@ -261,12 +258,7 @@ Projects that use *configure*, *cmake*, or some other portable configuration met
...
-
-The main limitation is that if you have multiple files in a single **.a** archive that have the same basename (for example, ``dir1/a.o, dir2/a.o``), then llvm-ar cannot access both of those files. Emscripten will attempt to work around this by adding a hash to the basename, but collisions are still possible in principle.
-
-Where possible it is better to generate shared library files (**.so**) rather than archives (**.a**) — this is generally a simple change in your project's build system. Shared libraries are simpler, and are more predictable with respect to linking.
-
+Emscripten supports **.a** archive files, which are bundles of object files. This is a simple format for libraries, that has special semantics - for example, the order of linking matters with **.a** files, but not with plain object files. For the most part those special semantics should work the same in Emscripten as elsewhere.
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:262]: maybe some notes about how to generate them? this came up in issues that I remember. like

```suggestion
To create **.a** archives, use ``emar``.
```

**Review by kripken — APPROVED** [review]
Maybe worth mentioning the output is executable by default now? I don't remember the exact change you made regarding that but there were some questions in an issue about it recently.

File: site/source/docs/compiling/Building-Projects.rst (hunk 1 above)
Comment (sbc100) [reply site/source/docs/compiling/Building-Projects.rst:262]: Its added and example to the list below.. is that not enough do you think?

**Comment by sbc100** [comment]
Added a specific note a bout the new default for linking.

File: site/source/docs/compiling/Building-Projects.rst [hunk 2]
```diff
@@ -72,23 +72,35 @@ To build with Emscripten, you would instead use the following commands:
...
+- ``emcc ... -o output.{html,js} -g4`` generates a source map file ``output.wasm.map``. If targeting asm.js with ``-s WASM=0``, the filename is ``output.{html,js}.map``.
+- ``emcc ... -o output.{html,js} --preload-file xxx`` directive generates a preloaded MEMFS filesystem file ``output.data``.
+- ``emcc ... -o output.{html,js} -s WASM={0,1} -s SINGLE_FILE=1`` merges JavaScript and WebAssembly code in the single output file ``output.{html,js}`` (in base64) to produce only one file for deployment. (If paired with ``--preload-file``, the preloaded ``.data`` file still exists as a separate file)
 
 This list is not exhaustive, but illustrates most commonly used combinations.
 
+.. note::
+   Regardless of the name of the output file ``emcc`` will always perform
+   linking and produce a final exectuable, unless a specific flags (e.g. ``-c``
+   direct it do something else).  This differs to previous beahvuiour where
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:99]:
```suggestion
   direct it do something else.  This differs to previous behaviour where
```

File: site/source/docs/compiling/Building-Projects.rst [hunk 3]
```diff
@@ -72,23 +72,35 @@ To build with Emscripten, you would instead use the following commands:
...
+- ``emcc ... -o output.{html,js} -s WASM=0 --memory-init-file 1`` causes the generation of ``output.{html,js}.mem`` memory initializer file. Pasing ``-O2``, ``-Os`` or ``-Oz`` also implies ``--memory-init-file 1``.
+- ``emcc ... -o output.{html,js} -g4`` generates a source map file ``output.wasm.map``. If targeting asm.js with ``-s WASM=0``, the filename is ``output.{html,js}.map``.
+- ``emcc ... -o output.{html,js} --preload-file xxx`` directive generates a preloaded MEMFS filesystem file ``output.data``.
+- ``emcc ... -o output.{html,js} -s WASM={0,1} -s SINGLE_FILE=1`` merges JavaScript and WebAssembly code in the single output file ``output.{html,js}`` (in base64) to produce only one file for deployment. (If paired with ``--preload-file``, the preloaded ``.data`` file still exists as a separate file)
 
 This list is not exhaustive, but illustrates most commonly used combinations.
 
+.. note::
+   Regardless of the name of the output file ``emcc`` will always perform
+   linking and produce a final exectuable, unless a specific flags (e.g. ``-c``
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:98]:
```suggestion
   linking and produce a final exectuable, unless a specific flags (e.g. ``-c``)
```

File: site/source/docs/compiling/Building-Projects.rst [hunk 4]
```diff
@@ -72,23 +72,35 @@ To build with Emscripten, you would instead use the following commands:
...
+   direct it do something else).  This differs to previous beahvuiour where
+   ``emcc`` would default to combining object files (essentailly assuming
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:100]:
```suggestion
   ``emcc`` would default to combining object files (essentially assuming
```

File: site/source/docs/compiling/Building-Projects.rst [hunk 5]
```diff
@@ -72,23 +72,35 @@ To build with Emscripten, you would instead use the following commands:
...
+   ``-r``) unless a spcific executable extansions (e.g. ``.js`` or ``.html``)
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:101]:
```suggestion
   ``-r``) unless given a specific executable extension (e.g. ``.js`` or ``.html``)
```

File: site/source/docs/compiling/Building-Projects.rst [hunk 6]
```diff
@@ -72,23 +72,35 @@ To build with Emscripten, you would instead use the following commands:
...
+Unless run with certain specific flags (such as ``-c``, ``-S``, ``-r``, or
+``-shared``) ``emcc`` will run the link phase which can produce more than just
+one file. The set of produced files changes depending on the final flags passed
+to ``emcc`` and the name of the specified output file. Here is a cheat sheet of
+which files are produced under which conditions:
+
+- ``emcc ... -o output.html`` builds a ``output.html`` file as an output, as well as an accompanying ``output.js`` launcher file, and a ``output.wasm`` WebAssembly file.
+- ``emcc ... -o output.js`` omits generating a HTML launcher file (expecting you to provide it yourself if you plan to run in browser), and produces two files, ``output.js`` and ``output.wasm``. (that can be run in e.g. node.js shell)
+- ``emcc ... -o output.wasm`` omits generating either JavaScript or HTML launcher file, and produces a single Wasm file built in standalone mode as if the ``-s STANDALONE_WASM`` settting had been used.
+- ``emcc ... -o output.{html,js} -s WASM=0`` causes the compiler to target asm.js, and therefore a ``.wasm`` file is not produced.
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:87]:
```suggestion
- ``emcc ... -o output.{html,js} -s WASM=0`` causes the compiler to target JavaScript, and therefore a ``.wasm`` file is not produced.
```

File: site/source/docs/compiling/Building-Projects.rst [hunk 7]
```diff
@@ -72,23 +72,35 @@ To build with Emscripten, you would instead use the following commands:
...
+- ``emcc ... -o output.{html,js} --emit-symbol-map`` produces a file ``output.{html,js}.symbols`` if WebAssembly is being targeted (``-s WASM=0`` not specified), or if asm.js is being targeted and ``-Os``, ``-Oz`` or ``-O2`` or higher is specified, but debug level setting is ``-g1`` or lower (i.e. if symbols minification did occur).
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:88]:
```suggestion
- ``emcc ... -o output.{html,js} --emit-symbol-map`` produces a file ``output.{html,js}.symbols`` if WebAssembly is being targeted (``-s WASM=0`` not specified), or if JavaScript is being targeted and ``-Os``, ``-Oz`` or ``-O2`` or higher is specified, but debug level setting is ``-g1`` or lower (i.e. if symbols minification did occur).
```

File: site/source/docs/compiling/Building-Projects.rst [hunk 8]
```diff
@@ -72,23 +72,35 @@ To build with Emscripten, you would instead use the following commands:
...
+- ``emcc ... -o output.{html,js} -s WASM=0 --memory-init-file 1`` causes the generation of ``output.{html,js}.mem`` memory initializer file. Pasing ``-O2``, ``-Os`` or ``-Oz`` also implies ``--memory-init-file 1``.
+- ``emcc ... -o output.{html,js} -g4`` generates a source map file ``output.wasm.map``. If targeting asm.js with ``-s WASM=0``, the filename is ``output.{html,js}.map``.
```
Comment (kripken) [inline site/source/docs/compiling/Building-Projects.rst:90]:
```suggestion
- ``emcc ... -o output.{html,js} -g4`` generates a source map file ``output.wasm.map``. If targeting JavaScript with ``-s WASM=0``, the filename is ``output.{html,js}.map``.
```

**Review by kripken — APPROVED** [review]
great!

**Pull request merged**